src/main/scala/io/iohk/ethereum/network/package.scala

```diff
@@ -31,8 +31,8 @@ package object network {
       val keysValuePair = generateKeyPair(secureRandom)
 
       //Write keys to file
-      val (_, priv) = keyPairToByteArrays(keysValuePair)
-      file.getParentFile.mkdirs()
+      val (priv, _) = keyPairToByteArrays(keysValuePair)
+      require(file.getParentFile.exists() || file.getParentFile.mkdirs(), "Key's file parent directory creation failed")
```

Comment:
I agree, `.mkdirs()` will work even if the dir already exits, however it will return `false`. So I added the `.exists()` check so as to distinguish the case when `.mkdirs()` failed and when the dir already existed